File: SDKs/googleapis.com/servicecontrol/v1/python/src/sdk/models/shared/tracespan.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import attributes as shared_attributes
from ..shared import truncatablestring as shared_truncatablestring
from ..shared import status as shared_status

class TraceSpanSpanKindEnum(str, Enum):
    SPAN_KIND_UNSPECIFIED = "SPAN_KIND_UNSPECIFIED"
    INTERNAL = "INTERNAL"
    SERVER = "SERVER"
    CLIENT = "CLIENT"
    PRODUCER = "PRODUCER"
    CONSUMER = "CONSUMER"


@dataclass_json
@dataclasses.dataclass
class TraceSpan:
    r"""TraceSpan
    A span represents a single operation within a trace. Spans can be nested to form a trace tree. Often, a trace contains a root span that describes the end-to-end latency, and one or more subspans for its sub-operations. A trace can also contain multiple root spans, or none at all. Spans do not need to be contiguous—there may be gaps or overlaps between spans in a trace.
    """
    
    attributes: Optional[shared_attributes.Attributes] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('attributes') }})
    child_span_count: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('childSpanCount') }})
    display_name: Optional[shared_truncatablestring.TruncatableString] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('displayName') }})
    end_time: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('endTime') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    parent_span_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('parentSpanId') }})
    same_process_as_parent_span: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sameProcessAsParentSpan') }})
    span_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('spanId') }})
    span_kind: Optional[TraceSpanSpanKindEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('spanKind') }})
    start_time: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('startTime') }})
    status: Optional[shared_status.Status] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    

File: SDKs/googleapis.com/servicecontrol/v1/java/lib/src/main/java/openapisdk/models/shared/TraceSpan.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * TraceSpan
 * A span represents a single operation within a trace. Spans can be nested to form a trace tree. Often, a trace contains a root span that describes the end-to-end latency, and one or more subspans for its sub-operations. A trace can also contain multiple root spans, or none at all. Spans do not need to be contiguous—there may be gaps or overlaps between spans in a trace.
**/
public class TraceSpan {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("attributes")
    public Attributes attributes;
    public TraceSpan withAttributes(Attributes attributes) {
        this.attributes = attributes;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("childSpanCount")
    public Integer childSpanCount;
    public TraceSpan withChildSpanCount(Integer childSpanCount) {
        this.childSpanCount = childSpanCount;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("displayName")
    public TruncatableString displayName;
    public TraceSpan withDisplayName(TruncatableString displayName) {
        this.displayName = displayName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("endTime")
    public String endTime;
    public TraceSpan withEndTime(String endTime) {
        this.endTime = endTime;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("name")
    public String name;
    public TraceSpan withName(String name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("parentSpanId")
    public String parentSpanId;
    public TraceSpan withParentSpanId(String parentSpanId) {
        this.parentSpanId = parentSpanId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("sameProcessAsParentSpan")
    public Boolean sameProcessAsParentSpan;
    public TraceSpan withSameProcessAsParentSpan(Boolean sameProcessAsParentSpan) {
        this.sameProcessAsParentSpan = sameProcessAsParentSpan;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("spanId")
    public String spanId;
    public TraceSpan withSpanId(String spanId) {
        this.spanId = spanId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("spanKind")
    public TraceSpanSpanKindEnum spanKind;
    public TraceSpan withSpanKind(TraceSpanSpanKindEnum spanKind) {
        this.spanKind = spanKind;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("startTime")
    public String startTime;
    public TraceSpan withStartTime(String startTime) {
        this.startTime = startTime;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("status")
    public Status status;
    public TraceSpan withStatus(Status status) {
        this.status = status;
        return this;
    }
}